lib/src/layer/polyline_layer.dart

```diff
@@ -54,71 +61,58 @@ class PolylineLayer extends StatelessWidget {
   /// performance of the layer.
   final bool saveLayers;
 
-  PolylineLayer({
+  const PolylineLayer({
     super.key,
     this.polylines = const [],
     this.polylineCulling = false,
     this.saveLayers = false,
-  }) {
-    if (polylineCulling) {
-      for (final polyline in polylines) {
-        polyline.boundingBox = LatLngBounds.fromPoints(polyline.points);
-      }
-    }
-  }
+  });
 
   @override
   Widget build(BuildContext context) {
     final map = FlutterMapState.maybeOf(context)!;
+    final zoom = map.zoom;
+    final rotation = map.rotation;
 
-    return LayoutBuilder(
-      builder: (BuildContext context, BoxConstraints bc) {
-        final size = Size(bc.maxWidth, bc.maxHeight);
-        final polylineWidgets = <Widget>[];
-
-        for (final polylineOpt in polylines) {
-          polylineOpt.offsets.clear();
-
-          if (polylineCulling &&
-              !polylineOpt.boundingBox.isOverlapping(map.bounds)) {
-            // skip this polyline as it's offscreen
-            continue;
-          }
-
-          _fillOffsets(polylineOpt.offsets, polylineOpt.points, map);
-
-          polylineWidgets.add(
-            CustomPaint(
-              key: polylineOpt.key,
-              painter: PolylinePainter(polylineOpt, saveLayers, map),
-              size: size,
-            ),
-          );
-        }
-
-        return Stack(
-          children: polylineWidgets,
-        );
-      },
-    );
-  }
+    final origin = map.pixelOrigin;
+    final offset = Offset(origin.x.toDouble(), origin.y.toDouble());
+
+    final polylineWidgets = <Widget>[];
+    for (final polyline in polylines) {
+      if (polylineCulling && !polyline.boundingBox.isOverlapping(map.bounds)) {
+        // Skip this polyline as it's offscreen
+        continue;
+      }
+
+      final paint = CustomPaint(
+        key: polyline.key,
+        painter: PolylinePainter(polyline, saveLayers, map, zoom, rotation),
+        size: Size(0, 0),
+        // If we were smarter we could set willChange to true during
+        // zooming/rotating the map and to false during moves.
+        willChange: true,
+        isComplex: true,
```

Comment:
Have you noticed if the raster caching resulting from `isComplex: true` suffer from lack of anti-aliasing just like when caching using a `RepaintBoundary`? It's usually not that glaring, but for geometries with thin strokes like polylines the jagged strokes became a bit glaring.

Reply:
I haven't noticed. Is there an example I can use to validate myself? Would you suggest to not raster cache at all or rather make it configurable?

Reply:
(just to avoid confusion, this PR is a bit stale. I've moved my own focus over to https://github.com/ignatz/flutter_map-clone/tree/fast_poly, which does both: raster caching but more importantly reduce the canvas draw operations (based on MooNag's proposal). The only reason I haven't closed this PR yet it because I was waiting for MooNag's input to tell me how they'd like to handle the PR)

Reply:
Feel free to continue wherever! I can close my PR once we merge yours in. If you need to make a new PR feel free to do so.

Reply:
SG. Let's keep the ball rolling. I opened https://github.com/fleaflet/flutter_map/pull/1442. And I will close this PR to avoid confusion.